src/components/AssetInfo/TokenSideBarRole.tsx

```diff
@@ -0,0 +1,26 @@
+import React from "react";
+
+interface TokenSideBarRoleProps {
+  adminAddress: string;
+  beneficiaryAddress: string;
+  holderAddress: string;
+}
+
+const TokenSideBarRole = ({ adminAddress, beneficiaryAddress, holderAddress }: TokenSideBarRoleProps) => {
+  const getUserRole = () => {
```

Comment:
We can reuse this logic in many place. Let's enum the state and maybe put this in the hook to get the user role directly.

Reply:
Updated.

Reply:
refacotred it.